packages/fetch-hooks/__tests__/hook.ts

```diff
@@ -97,4 +108,61 @@ describe("hook", () => {
             .toThrowError(new DuplicateInvocationError("FetchHookInvocation#next()"))
     })
 
+    test("start of hook chain is accessible", async () => {
+        /*
+         * How does this test work?
+         * 1. Two hooks are installed; requests go loopbackHook -> subRequestHook -> fakeFetch
+         * 2. From subRequestHook, send an additional request to the entrypoint. This should go through loopbackHook again
+         *      (which adds a header that is checked in fakeFetch)
+         * 3. From the response of that additional request, update the in-flight request and send it on
+         *
+         * We expect requests sent to the entrypoint to be round-tripped through all hooks again.
+         * If that does not happen, the X-Loopback: true header is not set on the request to example.com,
+         * which will respond with a 402 error. That will cause an error to be thrown without going to next
+         */
```

Comment:
I had to read this 3 times before I understood. Some things didn't help:

* "Two hooks are installed" yet there's clearly three in the code.
* The arrows mean processing in order, not sending additional requests that are handled by another hook.
* "The entrypoint" is just localhost, it's not referring to loopbackHook (which I thought was the entrypoint of the hook chain).
* "In-flight request" isn't referring to a request that's already in the process of doing http.

I reworded it a little to clarify the things I stumbled over. My wording probably has problems too, feel free to make it your own again.

```suggestion
        /*
         * The purpose of this test is to validate that when a hook sends a request of its own,
         * that request goes through all hooks again.
         *
         * How does this test work?
         * 1. Two hooks are installed; requests are processed by loopbackHook, then subRequestHook, then fakeFetch
         * 2. From subRequestHook, send an additional request to localhost. This should go through loopbackHook again
         *      (which adds a header that is checked in fakeFetch).
         * 3. From the response of that additional request, update the original request and send it on
         *
         * We expect requests sent to localhost to be round-tripped through all hooks again.
         * In order to validate that, we wrap the above situation in a third hook, the recursiveInvocationCheckHook.
         * If the request wouldn't again round-trip through all hooks, the X-Loopback: true header is not set on the request to example.com,
         * which will respond with a 402 error. That will cause an error to be thrown without going to next
         */
```

Reply:
> "Two hooks are installed" yet there's clearly three in the code.

There are two hooks, and one actual fetch function (`fakeFetch`). `fakeFetch` is not a hook, but a mock for `fetch`.
You also kept the description of "two hooks"

> "The entrypoint" is just localhost, it's not referring to loopbackHook (which I thought was the entrypoint of the hook chain).

I certainly meant `loopbackHook` (that is the entrypoint)

"Requests sent to localhost" is not really important for the explanation, and it depends on what you mean by "sent to" (is that executing the actual request, or calling the fetch entrypoint that still has to go through the hooks)